state/state.go

```diff
@@ -0,0 +1,91 @@
+package state
+
+import (
+	"log"
+	"sync"
+
+	"github.com/NordSecurity/nordvpn-linux/events"
+	"github.com/NordSecurity/nordvpn-linux/internal"
+)
+
+type Event int
+
+const (
+	VPNConnected Event = iota
+	VPNDisconnected
+)
+
+type subscriber struct {
+	stateChan chan<- Event
+	stopChan  <-chan struct{}
+}
+
+func newSubscriber(stateChan chan<- Event, stopChan <-chan struct{}) subscriber {
+	return subscriber{
+		stateChan: stateChan,
+		stopChan:  stopChan,
+	}
+}
+
+type subscriberSet map[subscriber]bool
+
+type StatePublisher struct {
+	mu          sync.Mutex
+	subscribers subscriberSet
+}
+
+func NewState() StatePublisher {
+	return StatePublisher{
+		subscribers: make(subscriberSet),
+	}
+}
+
+func (s *StatePublisher) notify(e Event) {
+	log.Println(internal.InfoPrefix+" received notification: ", e)
+	for sub := range s.subscribers {
+		select {
```

Comment:
shouldn't there be a `default` also in this case? if for some reason the event cannot be sent then it will remain blocked in this place?